### ISSUE TITLE
Add validations to Shelf model

### ISSUE THREAD

**Issue by xlisachan** [mozilla/addons-server#14952]
Supports #14871
Dependent on #14872 

Add validators to Shelf model that checks that `criteria` does not return 404s upon saving the instance.

Another issue will cover creating a validator for `footerPathname`.

**Comment by AlexandraMoga** [comment]
I've tried several `search` and `collections` combinations in the https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/shelfmodule/add/ tool and I can see the validation errors being thrown for the following combinations:
- for the `search` endpoint:
  - incorrect add-on `type` parameter
  - the missing `?`
  - incorrect `sort` parameters
- for the `collections` endpoint:
  - I can get the form to work if I'm using an existing mozilla collection, like `privacy-matters` 
  - I am receiving a form error if there is no mozilla collection with the input slug 

While testing this, I figured out it would be useful to have a preview of the full URL that result from these combinations in order to give the admins the possibility to live test the results on the frontend before Saving the shelf. This could be a field that starts from the base URL then it's updated based on the selected  'Endpoint' and the 'Criteria' parameters. @xlisachan does this make sense?

I've also filed #15282 as a follow up.

**Comment by xlisachan** [comment]
Thanks for your feedback! I think your suggestion will be added in a future issue, as this patch only covers validating the criteria and preventing the admin from saving the shelf if the results are invalid.

I have submitted a pull request for your follow-up. :)
